aksparse/client/aksparse/helpers.py

```diff
@@ -0,0 +1,71 @@
+import arkouda as ak
+
+def gen_ranges(starts, ends):
+    """ 
+    Generate a segmented array of variable-length, contiguous 
+    ranges between pairs of start- and end-points.
+
+    #TODO: Bring up what zero-filter does. Rename zero_filter.
+
+    Parameters
+    ----------
+    starts : pdarray, int64
+        The start value of each range
+    ends : pdarray, int64
+        The end value (exclusive) of each range
+
+    Returns
+    -------
+    segments : pdarray, int64
+        The starting index of each range in the resulting array
+    ranges : pdarray, int64
+        The actual ranges, flattened into a single array
+    """
+
+
+    lengths = ends - starts
+    segs = ak.cumsum(lengths) - lengths
+
+    zero_scan = segs[1:] - segs[:-1]
+    zero_filter = (zero_scan != 0)
+    zero_filter = ak.concatenate([zero_filter, ak.array([True])])
+
+    fsegs = segs[zero_filter]
+    fstarts = starts[zero_filter]
+    fends = ends[zero_filter]
+
+    totlen = lengths.sum()
+    slices = ak.ones(totlen, dtype=ak.int64)
+    diffs = ak.concatenate((ak.array([fstarts[0]]),
+                            fstarts[1:] - fends[:-1] +1))
+    slices[fsegs] = diffs
+    return fsegs, zero_filter, ak.cumsum(slices)
+
+def expand(vals, segs, size, zfilter):
```

Comment:
This function exists in `arkouda/util.py`. It looks like there may be a bit more functionality here, but we should probably make a decision to line these up with all required functionality in one place rather that duplicating.

Reply:
Adding a note here that possibly the best way to handle this would be to put an issue in both locations. One to update arkouda directly to include all the required functionality and one in contrib to update to use the arkouda function once the updates are made and remove the function from here.

Reply:
yeah some of the code here was done because we did want some additional functionality to faithfully represent matrix operations (I believe blank rows/columns was a big one).